datapath: Move node_config.h generation entirely to Go code

vadorovsky opened the pull request:
Before this change, the part of node_config.h with macros containing
information about Cilium network interfaces was generated in init.sh.
This change moves it to the part of Go codebase which is generating the
content of that file.

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

vadorovsky committed:
node: Move ephemeral port range getting into separate function

This change adds a new function `EphemeralPortRange` which gets the
range from sysctl.

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

vadorovsky committed:
datapath: Move node_config.h generation entirely to Go code

Before this change, the part of node_config.h with macros containing
information about Cilium network interfaces was generated in init.sh.
This change moves it to the part of Go codebase which is generating the
content of that file.

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

joestringer commented on pkg/datapath/linux/config/config.go:
```diff
@@ -204,6 +206,25 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 	cDefinesMap["TRACE_PAYLOAD_LEN"] = fmt.Sprintf("%dULL", option.Config.TracePayloadlen)
 	cDefinesMap["MTU"] = fmt.Sprintf("%d", cfg.MtuConfig.GetDeviceMTU())
 
+	ciliumHost, err := netlink.LinkByName("cilium_host")
```
Note, there's an ordering consideration here:
```
Daemon.init()` calls:
-> `d.createNodeConfigHeaderfile(...)`
    -> `d.datapath.WriteNodeConfig(...)`
-> ...
-> `d.Datapath().Loader().Reinitialize(...)`
    ->  `setupBaseDevice(...)`
```
In summary, the device may not exist until later.

vadorovsky replied:
I think I solved that with the last push and 1d0f6a80b9bd734a28cc3439eb923f14ebe6dd94. Thanks for pointing that out!

vadorovsky committed:
datapath: Create node config as Reinitialize() step

After previous change, node config generation expects base network
devices to exist. Thus move it to Reinitialize() function and call it
after creating those interfaces.

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>
